packages/opentelemetry/src/sampler.ts

```diff
@@ -67,6 +67,17 @@ export class SentrySampler implements Sampler {
     }
 
     const parentSampled = parentSpan ? getParentSampled(parentSpan, traceId, spanName) : undefined;
+    const isRootSpan = !parentSpan || parentContext?.isRemote;
+
+    // We only sample based on parameters (like tracesSampleRate or tracesSampler) for root spans (which is done in sampleSpan).
+    // Non-root-spans simply inherit the sampling decision from their parent.
+    if (!isRootSpan) {
+      return wrapSamplingDecision({
```

Comment:
Don't we have to run `this._client.emit('beforeSampling'` for this as well?

Reply:
I don't think so, we only use this (today) in Next.js, where I assume this is for root spans:

```js
// There are situations where the Next.js Node.js server forwards requests for the Edge Runtime server (e.g. in
    // middleware) and this causes spans for Sentry ingest requests to be created. These are not exempt from our tracing
    // because we didn't get the chance to do `suppressTracing`, since this happens outside of userland.
    // We need to drop these spans.
```

cc @lforst ?

Reply:
I mean all the tests are passing, I would assume something would fail if this breaks it :) (and sampling should not be applied to child spans anyhow, so this should be fine!)